src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java

```diff
@@ -267,38 +279,39 @@ public void preSubmissionCheck() {
         }
     }
 
-    private void postSettlement() {
+    private void postSettlement() throws IOException, InterruptedException {
         try {
             getPauseAction().remove(this);
-            run.save();
+            getRun().save();
         } catch (IOException | InterruptedException | TimeoutException x) {
-            LOGGER.log(Level.WARNING, "failed to remove InputAction from " + run, x);
+            LOGGER.log(Level.WARNING, "failed to remove InputAction from " + getContext(), x);
```

Comment:
Looks reasonable, but here and elsewhere where you call `getContext()` did you mean `getRun()`? That would make more sense to me as a direct translation of the old `run` variable.

Reply:
(I see what you mean about GitHub not letting you comment on a single line without leaving a "review". I clicked "leave a single comment", yet now it appears as "reviewed 18 minutes ago.")

Reply:
`getRun()` would be the more direct equivalent, but might throw exceptions. `getContext()` is good enough for logging purposes as its `toString()` will include an identification of the build: https://github.com/jenkinsci/workflow-cps-plugin/blob/a9433432b33c65f58d9e7b4dc4aa83249672eaf7/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsStepContext.java#L596-L605 + https://github.com/jenkinsci/workflow-job-plugin/blob/969637446d3ba7ea2c1b26cb7ee1b9c44c6e4ebc/src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java#L1003-L1005

Reply:
Got it, sounds good then. Resolving.